experimental/uefi/runtime/src/framing.rs

```diff
@@ -25,69 +28,109 @@ use oak_remote_attestation::handshaker::{
 };
 use oak_remote_attestation_sessions::SessionId;
 
-struct InvocationHandler<F, G, V>
+enum AttestationState<G, V>
+where
+    G: AttestationGenerator,
+    V: AttestationVerifier,
+{
+    Uninitialized(AttestationBehavior<G, V>),
+    Initialized(Box<dyn AttestationTrait>),
+}
+
+struct InvocationHandler<G, V>
 where
-    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> anyhow::Result<Vec<u8>>,
     G: AttestationGenerator,
     V: AttestationVerifier,
 {
-    attestation_handler: AttestationHandler<F, G, V>,
+    attestation_state: AttestationState<G, V>,
 }
 
-impl<F, G, V> schema::TrustedRuntime for InvocationHandler<F, G, V>
+impl<G: 'static, V: 'static> schema::TrustedRuntime for InvocationHandler<G, V>
 where
-    F: Send + Sync + Clone + FnOnce(Vec<u8>) -> anyhow::Result<Vec<u8>>,
     G: AttestationGenerator,
     V: AttestationVerifier,
 {
+    fn initialize(
+        &mut self,
+        initialization: &schema::Initialization,
+    ) -> Result<oak_idl::utils::Message<channel::schema::Empty>, oak_idl::Status> {
+        match &mut self.attestation_state {
+            AttestationState::Initialized(_attestation_handler) => Err(oak_idl::Status::new(
+                oak_idl::StatusCode::FailedPrecondition,
+            )),
+            AttestationState::Uninitialized(attestation_behavior) => {
+                let wasm_module_bytes: &[u8] = initialization
+                    .wasm_module()
+                    .ok_or_else(|| oak_idl::Status::new(oak_idl::StatusCode::InvalidArgument))?;
+                let wasm_handler = wasm::new_wasm_handler(wasm_module_bytes)
+                    .map_err(|_| oak_idl::Status::new(oak_idl::StatusCode::Internal))?;
+                let attestation_handler = Box::new(AttestationHandler::create(
+                    move |v| wasm_handler.handle_raw_invoke(v),
+                    attestation_behavior.clone(),
```

Comment:
Rust question!

I'm not a huge fan of this `.clone` operation, since the `attestation_behavior` is no longer needed after it has been used here.

However, rust doesn't let me take it, since it's behind a mutable reference. 

Is there a way I could avoid this operation? I looked at storing `self.attestation_state` in a `RefCell` and calling `replace_with`, but it also doesn't allow me to consume the previous value.

Reply:
Since you already modified the bounds to include a 'static lifetime (which I think makes sense), then perhaps we can just a reference to the attestation behaviour from everywhere, so that it never needs to be cloned at all. Would that work?